app/controllers/records_controller.rb

```diff
@@ -19,6 +19,37 @@ def refresh
     redirect_to (request.referrer || home_path)
   end
 
+  def associate_granule_to_collection
+    @record = Record.find_by id:params[:id]
+    collection = Collection.find_by id: @record.recordable_id
+    associated_granule_value = params[:associated_granule_value]
+
+    if associated_granule_value == 'Undefined'
+      flash[:notice] = "This revision #{@record.revision_id} associated granule will be marked as undefined."
+      @record.associated_granule_value = nil
+    elsif associated_granule_value == 'No Granule Review'
+      flash[:notice] = "This revision #{@record.revision_id} associated granule will be marked as 'No Granule Review'"
+      @record.associated_granule_value = associated_granule_value
+    elsif associated_granule_value == 'Granule Review Deleted!'
+      flash[:notice] = "This revision #{@record.revision_id} associated granule is marked as 'Granule Review Deleted!"
+      @record.associated_granule_value = associated_granule_value
+    else
+      granule_record = Record.find_by id: associated_granule_value
+      if !granule_record.nil?
+        granule = Granule.find_by id: granule_record.recordable_id
+        granule_concept_id = granule.concept_id
+        granule_revision_id = granule_record.revision_id
+        flash[:notice] = "Granule #{granule_concept_id}/#{granule_revision_id} has been successfully associated to this collection revision #{@record.revision_id}. "
+        @record.associated_granule_value = granule_record.id
```

Comment:
does the 'association' happen before or after this assignment, and before or after the `save!` below?

Reply:
Not sure I understand, it happens before, as the save! is after in code, but I'm sure you mean something else.

Reply:
The latest code should be much cleaner here.